### PR TITLE
Port command-error-or-default-function

### PR DESCRIPTION
**Pull request by krobelus**
This requires conditional compilation based on whether the platform is windows (for `is_daemon()`).
Not sure what's the best way to do that. I think this works but I didn't test it on windows.

### CODE REVIEW

**Comment by agraven** [comment]
I think `#[cfg(windows)]` should work for that

File: rust_src/Cargo.toml.in [hunk 1]
```diff
@@ -63,6 +63,8 @@ window-system-x11 = []
 window-system-nextstep = []
 # Use the w32 window system
 window-system-w32 = []
+# Compiling for windows
```
Comment (shaleh) [inline rust_src/Cargo.toml.in]: As @agraven points out, there is no need for this. Rust already defines `cfg(windows)`.

File: rust_src/src/emacs.rs [hunk 2]
```diff
@@ -21,4 +26,11 @@ pub fn invocation_directory() -> LispObject {
     unsafe { Fcopy_sequence(globals.Vinvocation_directory) }
 }
 
+pub fn is_daemon() -> bool {
+    #[cfg(feature = "windowsnt")]
```
Comment (shaleh) [inline rust_src/src/emacs.rs]: This is a good use for the `cfg-if` crate.

File: rust_src/src/keyboard.rs [hunk 3]
```diff
@@ -4,21 +4,30 @@ use remacs_macros::lisp_fn;
...
+    emacs::is_daemon,
     eval::{record_unwind_protect, unbind_to},
     frames::{selected_frame, window_frame_live_or_selected_with_action},
     lisp::defsubr,
     lisp::LispObject,
     lists::{LispCons, LispConsCircularChecks, LispConsEndChecks},
+    multibyte::LispStringRef,
     numbers::IsLispNatnum,
+    remacs_sys::globals,
+    remacs_sys::{clear_message, message_log_maybe_newline, print_error_message},
```
Comment (shaleh) [inline rust_src/src/keyboard.rs]: Add these to the existing function imports below.

File: rust_src/src/keyboard.rs [hunk 4]
```diff
@@ -220,4 +229,52 @@ pub extern "C" fn rust_syms_of_keyboard() {
...
 
+/// Produce default output for unhandled error message.
+/// Default value of `command-error-function'.
+#[lisp_fn]
+pub fn command_error_default_function(
+    data: LispObject,
+    context: LispStringRef,
+    signal: LispObject,
+) -> LispObject {
+    let sf = selected_frame();
```
Comment (shaleh) [inline rust_src/src/keyboard.rs]: Let's try to use better variable names in Remacs.

File: rust_src/src/keyboard.rs [hunk 5]
```diff
@@ -220,4 +229,52 @@ pub extern "C" fn rust_syms_of_keyboard() {
...
+    } else {
+        unsafe {
+            clear_message(true, false);
+            Fdiscard_input();
+            message_log_maybe_newline();
+            ding_internal(true);
+            print_error_message(data, Qt, context.const_sdata_ptr(), signal);
+        }
+    }
+    Qnil
```
Comment (shaleh) [inline rust_src/src/keyboard.rs]: There is no need for the plain `Qnil` return. Leave off the return type in the function definition and the right thing will happen.

**Comment by daichimukai** [comment]
:tada:

**Pull request merged**